### DISCUSSION TITLE
Plugins usage and documentation

### DISCUSSION THREAD

**Discussion by AnthonyJZuniga** [ColorlibHQ/AdminLTE#3788] [Q&A] [answered]
From what I understand installing adminlte via NPM includes plugins like datatables. However it's not clear exactly how to include and use them in, for example, a laravel project. Is there something I'm missing here? I know datatables has documentation for installing it standalone via NPM but specifically using what is included with adminlte is a mystery to me. What do I need to add to my app.js and app.scss to make these plugins work? The adminlte documentation includes a page showing datatables working but does not explain how to configure it to work without a CDN.

**Comment by AnthonyJZuniga** [comment, marked as answer]
After some trial and error I think I figured it out. The following will be using Datatables and some of its extensions that I'm using in a Laravel 7 project.

In resources/js/app.js:
`
// Bootstrap
require('./bootstrap');

// AdminLTE
require('admin-lte');
require('admin-lte/plugins/datatables-bs4/js/dataTables.bootstrap4');
require('admin-lte/plugins/datatables-select/js/select.bootstrap4');
require('admin-lte/plugins/datatables-scroller/js/scroller.bootstrap4');
require('admin-lte/plugins/datatables-buttons/js/buttons.bootstrap4');
require('admin-lte/plugins/datatables-buttons/js/buttons.colVis.js');
require('admin-lte/plugins/datatables-buttons/js/buttons.html5.js');
require('admin-lte/plugins/datatables-buttons/js/buttons.flash.js');
require('admin-lte/plugins/datatables-buttons/js/buttons.print.js');
`

In resources/sass/app.scss:
`
// Variables
@import 'variables';

// AdminLTE
@import '~admin-lte/build/scss/AdminLTE';
@import '~admin-lte/plugins/datatables-bs4/css/dataTables.bootstrap4';
@import '~admin-lte/plugins/datatables-select/css/select.bootstrap4';
@import '~admin-lte/plugins/datatables-scroller/css/scroller.bootstrap4';
@import '~admin-lte/plugins/datatables-buttons/css/buttons.bootstrap4';
`

Basically import all scss files for the plugin in app.scss and require all js files in app.js. Check node_modules/admin-lte/plugins for file paths necessary for the plugin you want to use.
